From 25c44ae05432f2a3b7389974ae395ed77333aafe Mon Sep 17 00:00:00 2001
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Tue, 10 Jun 2025 12:42:02 +0200
Subject: [PATCH 22/22] CryptoPkg/openssl: disable some features support

Reduce library size by turning off:
 * quic protocol support.
 * post-quantum chipers.

Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
---
 CryptoPkg/Library/OpensslLib/configure.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/CryptoPkg/Library/OpensslLib/configure.py b/CryptoPkg/Library/OpensslLib/configure.py
index 37e5cf395da2..beb6f3ed61b6 100755
--- a/CryptoPkg/Library/OpensslLib/configure.py
+++ b/CryptoPkg/Library/OpensslLib/configure.py
@@ -53,6 +53,8 @@ def openssl_configure(openssldir, target, ec = True):
         'no-module',
         'no-md4',
         'no-mdc2',
+        'no-ml-dsa',
+        'no-ml-kem',
         'no-multiblock',
         'no-nextprotoneg',
         'no-pic',
@@ -62,6 +64,7 @@ def openssl_configure(openssldir, target, ec = True):
         'no-padlockeng',
         'no-poly1305',
         'no-posix-io',
+        'no-quic',
         'no-rc2',
         'no-rc4',
         'no-rc5',
@@ -72,6 +75,7 @@ def openssl_configure(openssldir, target, ec = True):
         'no-shared',
         'no-siphash',
         'no-siv',
+        'no-slh-dsa',
         'no-sm2',
         'no-sm4',
         'no-sock',
-- 
2.49.0

